### PR TITLE
Make headless services support in multicluster link chart configurabl…

### LINKED ISSUE #6562: Add `enableHeadlessServices` flag to multicluster extension
In PR #6090 (_"Introduce support for StatefulSet across clusters"_) we introduced a new [`--enable-headless-services`](https://github.com/linkerd/linkerd2/blob/efeb21a2a9871af1d39f1e34475d59e8db62dd55/multicluster/cmd/service-mirror/main.go#L42) flag for the service mirror to allow headless services to be exported from a target cluster to a source cluster.

At the moment, in order to turn the feature on, we have to manually edit the output from `linkerd multicluster link` to add the flag to the service-mirror controller. This should be supported as an actual value in the helm chart so we can more easily install multicluster with headless services support.

I think the easiest way to do this would be to have a `enableHeadlessServices` field in the [multicluster link chart](https://github.com/linkerd/linkerd2/blob/main/multicluster/charts/linkerd-multicluster-link/values.yaml). If the value of the field is set to `true`, we can add the argument in the service-mirror deployment (it would be an argument to the main container). Users can then make use of the `--set` flag on the cli to turn this on; e.g `linkerd multicluster link --cluster <target> --set "enableHeadlessServices=true"`.

### PR DESCRIPTION
**Pull request by knutgoetz**
Running `linkerd multicluster link --cluster <target> --set "enableHeadlessServices=true"`
should enable the [headless service functionality](https://github.com/linkerd/linkerd2/pull/6090).

Adds the `enableHeadlessServices` flag to the helm chart, which adds the
`-enable-headless-services` container argument to the controller.

Just tested the rendering of the chart. Will test using the chart in a cluster now.

Fixes #6562

Signed-off-by: Knut Götz <knutgoetz@gmail.com>

### CODE REVIEW

**Comment by mateiidavid** [comment]
Hey @knutgoetz, thanks for putting this up! If you haven't already seen this, we have a [repository](https://github.com/mateiidavid/l2d-k3d-statefulset) with some test scripts to spin up multicluster on your local machine. 

You'll want to avoid using the `link.sh` script and instead link the clusters using the cli -- the script itself `sed`s the flag in the service-mirror's container args, so we'll want to do it without that.

File: multicluster/charts/linkerd-multicluster-link/values.yaml [hunk 1]
```diff
@@ -3,6 +3,8 @@
 controllerImage: cr.l5d.io/linkerd/controller
 # -- Tag for the Service Mirror container Docker image
 controllerImageVersion: linkerdVersionValue
+# -- Toogle support for mirroring headless services
```
Comment (kleimkuhler) [inline multicluster/charts/linkerd-multicluster-link/values.yaml:6]:
```suggestion
# -- Toggle support for mirroring headless services
```

**Review by kleimkuhler — COMMENTED** [review]
You'll probably have to run `bin/helm-docs` to make sure the generated Helm docs get updated with the addition of the new flag.

File: multicluster/charts/linkerd-multicluster-link/values.yaml (hunk 1 above)
Comment (knutgoetz) [reply multicluster/charts/linkerd-multicluster-link/values.yaml:6]: Ah :) Thx. And I'll run the `helm-docs`

**Review by kleimkuhler — APPROVED** [review]
Thanks @knutgoetz!

**Review by mateiidavid — APPROVED** [review]
Awesome work @knutgoetz! 🚀

**Pull request merged**